[web-app] page load improvements

jharvey10 opened the pull request:
- Improve page load times by using dynamic imports for large chunks (pictograms and icons metadata)
- Fix some typos
- Remove library provider and context in favor of a prop
- Remove IDL-site conditional from icons action bar

Fixes #1201

#### Testing / reviewing

- Ensure feature parity between old icons and pictograms pages and new ones
- Run a production build of the web-app `npm -w services/web-app run build`, run the production build locally `npm -w services/web-app run start`, and ensure all pages load faster than before (depending on network speed)

jharvey10 committed:
fix(web-app): dynamically import icons and pictograms

Fixes #1201

- Improve page load times by using dynamic imports for large chunks (pictograms and icons metadata)
- Fix some typos
- Remove library provider and context in favor of a prop
- Remove IDL-site conditional from icons action bar

alisonjoseph reviewed (changes requested):
Looks like we lost the correct padding at the top of the page when switching from mdx to js page. Wondering if we should move this style to a global wrapper, although may add incorrect spacing to other pages if we do that. Can probably look at that in a different issue unless its an easy fix.

![Screen Shot 2022-09-01 at 9 56 19 AM](https://user-images.githubusercontent.com/2753488/187946150-2101a2d9-3239-4ac4-b22e-536fde8b6251.png)

![Screen Shot 2022-09-01 at 9 58 20 AM](https://user-images.githubusercontent.com/2753488/187946385-6f75d445-d4a0-402b-b7a8-1d18e4af80e6.png)


Everything else is running / working great!

jdharvey-ibm commented:
@alisonjoseph Fixed! Will upload it shortly. For now, since there was already a class applying `spacing-07`, I just changed it to `spacing-11`, but we can definitely move it to a top-level style. I'd just probably need help figuring out all the places to remove existing padding in favor of that one. (Not sure if it's just MDX pages or if there are other ones that would benefit from that as well)

jharvey10 committed:
fix(web-app): correct spacing at top of svg library pages

alisonjoseph commented:
@jdharvey-ibm yea we should probably make a separate issue to look into this, I think its going to be a bit of a game of whack-a-mole, and syncing with design to make sure we have consistent top spacing in the correct places.

jharvey10 committed:
refactor(web-app): reduce lodash footprint

- use lodash subdirectory imports for better tree shaking at build time
- remove most usages of `get`

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into fix/1201

jharvey10 committed:
refactor(web-app): reduce assets index cognitive complexity

francinelucca commented on services/web-app/pages/elements/icons/library.js:
```diff
@@ -0,0 +1,76 @@
+  )
+}
+
+export const getStaticProps = async () => {
```
you could use getStaticProps inside the .mdx file (see services/web-app/pages/designing/get-started.mdx), is there a reason you chose to convert to .js file instead? (just curious here)

francinelucca reviewed (changes requested):
UI and build LGTM
PictogramLibrary and IconLibrary storybook no longer working for me

francinelucca commented:
forgot to attach pic 
<img width="1790" alt="image" src="https://user-images.githubusercontent.com/40550942/188187861-66c42f07-6b68-41af-86cc-5c8430f86265.png">

jharvey10 committed:
fix: revert back to mdx icon and pictogram libraries

- revert top-padding adjustment
- remove icons/pictograms stories in favor of eventual static export-based VRT

francinelucca reviewed (approved):
LGTM
